cli/src/main/java/com/dylibso/chicory/cli/Cli.java

```diff
@@ -66,31 +68,19 @@ public void run() {
                                         .toHostFunctions())
                         : new HostImports();
         var instance =
-                Module.builder(file)
-                        .withLogger(logger)
+                Instance.builder(module)
                         .withInitialize(true)
                         .withStart(false)
                         .withHostImports(imports)
-                        .build()
-                        .instantiate();
+                        .build();
 
         if (functionName != null) {
-            var exportSig = module.export(functionName);
             var export = instance.export(functionName);
-
-            var typeId = instance.functionType(exportSig.index());
-            var type = instance.type(typeId);
-
-            if (arguments.length != type.params().size()) {
-                throw new RuntimeException(
-                        "The function needs "
-                                + type.params().size()
-                                + " parameters, but found: "
-                                + arguments.length);
-            }
-            var params = new Value[type.params().size()];
-            for (var i = 0; i < type.params().size(); i++) {
-                params[i] = new Value(type.params().get(i), Long.valueOf(arguments[i]));
+            var params = new Value[arguments.length];
+            for (var i = 0; i < arguments.length; i++) {
+                // TODO: FIXME - it's hard to extract the exported function signature
```

Comment:
this is the only "rough edge" that, in my opinion, is left from this PR.

Reply:
The validation is performed while building an `Instance`, the function signatures are accessible in the raw `wasm.Module`, just not as convenient as it used to be.

Reply:
Added an `exportType` method to fill the gap.